[Core] Update to OpenAssetIO 1.0.0-beta.1.0

feltech opened the pull request:
Update dependency versions, hence

* Avoid use of deprecated exceptions.
* Explicitly detect when an entity has no "locatableContent" trait. Previously an exception would be raised from OpenAssetIO, but this was changed to return `None` instead.

feltech committed:
[Core] Update to OpenAssetIO 1.0.0-beta.1.0

Update dependency versions, hence

* Avoid use of deprecated exceptions.
* Explicitly detect when an entity has no "locatableContent" trait.
  Previously an exception would be raised from OpenAssetIO, but this
  was changed to return `None` instead.

Signed-off-by: David Feltell <david.feltell@foundry.com>

foundrytom commented on setup.py:
```diff
@@ -28,7 +28,7 @@
     install_requires=[
         "OpenTimelineIO >= 0.12.0",
-        "openassetio == 1.0.0a14",
+        "openassetio == 1.0.0b1.rev0",
```
[consider] We always wanted to avoid hard pins, and use semver (not sure how white space works here)
```suggestion
        "openassetio == 1.0.0b1.rev0, < 1.0.0b2.rev0",
```

feltech replied:
Updated as suggested in 300fa844

foundrytom reviewed (commented):
Generally looks good to me minus linter fixes.

foundrytom commented on otio_openassetio/operations/openassetio_media_linker.py:
```diff
@@ -7,7 +7,7 @@
 from collections import namedtuple
 
-from openassetio import log, exceptions
+from openassetio import log, errors
```
[Fix] Be good to add a `hasCapability` check at L165

feltech replied:
Little bit of linting in 1181013b then added the check in 5281e963

foundrytom replied:
Sorry, floating comment [here](https://github.com/OpenAssetIO/otio-openassetio/commit/5281e963b4285ef34c371405f0fe044dc2e3b193#r130680493)

feltech replied:
^ updated exception in 3b929a98

feltech committed:
[Lint] Black format test module

Signed-off-by: David Feltell <david.feltell@foundry.com>

feltech committed:
[Core] Add `hasCapability` check

Use the new `hasCapability` API to assert that the desired manager will
be able to `resolve` entity references.

Signed-off-by: David Feltell <david.feltell@foundry.com>

foundrytom commented:
Sorry GitHub has done one of those floating comments again [here](https://github.com/OpenAssetIO/otio-openassetio/commit/5281e963b4285ef34c371405f0fe044dc2e3b193#r130680493).

foundrytom commented on tests/test_otio_openassetio.py:
```diff
@@ -162,9 +185,7 @@ def test_when_no_manager_setting_and_no_default_config_then_error_raised(
-    with pytest.raises(
-        RuntimeError, match="No default OpenAssetIO manager configured"
-    ):
+    with pytest.raises(RuntimeError, match="No default OpenAssetIO manager configured"):
```
[Question] Whilst were here, should this also be a `ConfigurationError`? Sorry missed this before (or make the other a RuntimeError so it's consistent?)

feltech replied:
Good spot. Used `ConfigurationError` in 8bd8066e.

There is a broader question of who should use OpenAssetIO exceptions.  I.e. host integration points, middleware like this OTIO plugin (perhaps differentiating between OpenAssetIO ecosystem vs. third-party), and manager implementations - vs. internal OpenAssetIO functions.  Would be good to come up with some guidance.

foundrytom replied:
Yeah 🤔 any thoughts on a starting point?

feltech replied:
As a starting point
* Host integration points - depends on how loosely coupled.  If a certain app has its own decoupled asset API, they should use OpenAssetIO exceptions where its an OpenAssetIO-specific issue.
* Middleware (ours) - use OpenAssetIO exceptions
* Middleware (third-party) - context-dependent, but probably same advice as for host integration points
* Managers - use OpenAssetIO exceptions

foundrytom replied:
https://github.com/OpenAssetIO/OpenAssetIO/issues/1184
